build_tools/cmake/iree_macros.cmake

```diff
@@ -514,6 +514,37 @@ function(iree_symlink_tool)
   )
 endfunction()
 
+#-------------------------------------------------------------------------------
+# Test timeout values
+#-------------------------------------------------------------------------------
+
+# iree_get_timeout_seconds()
+#
+# Evaluates a Bazel-like timeout string to a number of seconds.
+# Allowed timeout values:  "short", "moderate", "long".
+# Default is "short".
+#
+# This is aligned with our `native_test` function in Bazel. This differs from
+# general Bazel rules in two ways: they default to "moderate" and they allow an
+# additional timeout value "eternal".
+#
+# Note: the current numbers of seconds are just the Bazel standard values.
+# They could change in the future. We don't actually want "short" tests that
+# run for 60 seconds, and we might never want any test to run for 900 seconds.
+# The main purpose of these timeout enumeration values is to filter and skip
+# certain tests on slow configurations, not to allow tests to run for a long
+# time.
+function(iree_get_timeout_seconds DST_VAR SRC_TIMEOUT_STRING)
```

Comment:
I had been imagining that these would resolve to CMake variables that we could then set how we'd like. My thought was that when translating Bazel->CMake we would set

```cmake
TIMEOUT
  "${IREE_TEST_TIMEOUT_SHORT}"
```

or something like that. Doing the translation in a function also works and we could put the variables here as well. Advantage is that it makes the translation from Bazel a bit simpler and the declaration of an individual test less verbose. The custom mapping seems a bit more "magic" though. IDK this seems fine now that I see it, but wanted to raise that there's another way, in case you hadn't considered that.